core/desugarSessionExceptions.ml

```diff
@@ -87,7 +87,8 @@ object (o : 'self_type)
         let ty =
           Types.fresh_type_variable (CommonTypes.lin_any, CommonTypes.res_any) in
         let with_pos x = SourceCode.WithPos.make ~pos x in
-        let doOp = DoOperation (with_pos (Operation failure_op_name), [], Some (Types.empty_type)) in
+        (* FIXME: WT: I don't know whether should it be lindo or not *)
```

Comment:
Though, I see that you have changed the signature of `SessionFail` below... I would be inclined not to change any of the session exceptions stuff.

Reply:
I have to change the `SessionFail` operation to be linear to make the functions in `prelude.links` well-typed. Otherwise we would get a type error because we use the unlimited operation `SessionFail` with linear channels. I also need to change some the signatures of some functions in `lib.ml`, because by default `e::Row` is `e::Row(Unl,Any)`, which can only be unified with linear operations.

This is also a challenge for hiding this feature behind a new flag; it is not entirely compatible with the current Links syntax.

Reply:
I see... The settings logic does allow you specify transitive dependencies, so if session exceptions are toggled, then you can forcibly toggle linear continuations too.

Reply:
Actually it is fine to not toggle control_flow_linearity, but we still need to make sure that the existing library code is compatible with it. FOr example, using unlimited row variable `e::Row(Any, Any)` by default and use the linear operation signature `=@` for `SessionFail`.